opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/TopologyRestService.java

```diff
@@ -0,0 +1,142 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2018 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.web.rest.v1;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.Response;
+
+
+
+import org.graphdrawing.graphml.GraphmlType;
+import org.opennms.features.graphml.model.GraphML;
+import org.opennms.features.graphml.model.GraphMLEdge;
+import org.opennms.features.graphml.model.GraphMLGraph;
+import org.opennms.features.graphml.model.GraphMLNode;
+import org.opennms.features.graphml.model.GraphMLWriter;
+import org.opennms.features.graphml.model.InvalidGraphException;
+import org.opennms.netmgt.dao.api.TopologyDao;
+import org.opennms.netmgt.model.OnmsTopology;
+import org.opennms.netmgt.model.topology.Topology;
+import org.opennms.netmgt.model.topology.Topology.ProtocolSupported;
+import org.springframework.stereotype.Component;
+
+@Component
+@Path("topology")
+public class TopologyRestService {
```

Comment:
Can we write this a bit more generic?
At the moment this only allows us to export topologies defined by linkd, but I'd prefer a version which would also simply allow us to export any other existing topologies, e.g. bsm, graphml, etc. 

For this, I'd create a `TopologyProvider` (maybe a better name) which for now returns a `GraphMLType` allowing us to export more topologies.

Reply:
Markus I agree but consider that the issue is create a CDP topology REST Service.

Reply:
I moved the path to topologies....eheh
Also consider that for some reason I do not understant the path rest/topology/CDP will be intercepted by topology app and the topology map is displayed....ehehe

Reply:
The Service is quite abstract.
You can also registrer a graphML updaters...and do updates via that service

The name is a convention.

So for example your graphml REST service will provide CRUD fir graphml topologies.

You can get a new graphml say myspecialtopology.
Then you subscribe the TopologyService for the name GrapPHML/myspecialTopology

Then the TopologyProvider will ask the TopologyService to get the GrapPHML/myspecialtopology and get the right graph